shell/platform/fuchsia/flutter/vulkan_surface.cc

```diff
@@ -15,52 +17,51 @@
 #include "third_party/skia/include/gpu/GrBackendSurface.h"
 #include "third_party/skia/include/gpu/GrDirectContext.h"
 
+#define LOG_AND_RETURN(cond, msg) \
+  if (cond) {                     \
+    FML_DLOG(ERROR) << msg;       \
+    return false;                 \
+  }
+
 namespace flutter_runner {
 
 namespace {
 
-// Immutable format is technically limited to R8G8B8A8_SRGB but
-// R8G8B8A8_UNORM works with existing ARM drivers so we allow that
-// until we have a more reliable API for creating external Vulkan
-// images using sysmem. TODO(fxb/52835)
 #if defined(__aarch64__)
 constexpr SkColorType kSkiaColorType = kRGBA_8888_SkColorType;
-constexpr fuchsia::images::PixelFormat kPixelFormat =
-    fuchsia::images::PixelFormat::R8G8B8A8;
+constexpr fuchsia::sysmem::PixelFormatType kSysmemPixelFormat =
+    fuchsia::sysmem::PixelFormatType::R8G8B8A8;
 constexpr VkFormat kVulkanFormat = VK_FORMAT_R8G8B8A8_UNORM;
 constexpr VkImageCreateFlags kVulkanImageCreateFlags = 0;
 #else
 constexpr SkColorType kSkiaColorType = kBGRA_8888_SkColorType;
-constexpr fuchsia::images::PixelFormat kPixelFormat =
-    fuchsia::images::PixelFormat::BGRA_8;
+constexpr fuchsia::sysmem::PixelFormatType kSysmemPixelFormat =
+    fuchsia::sysmem::PixelFormatType::BGRA32;
 constexpr VkFormat kVulkanFormat = VK_FORMAT_B8G8R8A8_UNORM;
 constexpr VkImageCreateFlags kVulkanImageCreateFlags =
     VK_IMAGE_CREATE_MUTABLE_FORMAT_BIT;
 #endif
```

Comment:
Can we remove this ifdef now?

Reply:
We should keep the else branch, right?

Reply:
No, else branch is what we want to remove. If branch is required for AFBC so critical to keep. The else branch was here to match intel image import details in Scenic but sysmem should remove the need for this.